src/foam/dao/RestDAO.js

```diff
@@ -87,6 +88,14 @@ foam.CLASS({
        * Each key's value is network-foam-jsonified.
        */
       var query = [];
+
+      // Proxy non-serializable sinks. Otherwise, serialize and send the sink,
+      // then instantiate the results when they come back from the server.
+      var proxySink = this.Serializable.isInstance(sink) ? null :
```

Comment:
Wouldn't this work (renaming to networkSink)?

var networkSink = this.Serializable.isInstance(sink) && sink;